lib/firebase_functions_interop.dart

```diff
@@ -378,3 +382,78 @@ class FirestoreEvent extends Event<DeltaDocumentSnapshot> {
           timestamp: timestamp,
         );
 }
+
+class PubsubFunctions {
+  const PubsubFunctions._();
+
+  TopicBuilder topic(String path) => new TopicBuilder._(_js.pubsub.topic(path));
+}
+
+class TopicBuilder {
+  @protected
+  final js.TopicBuilder nativeInstance;
+
+  TopicBuilder._(this.nativeInstance);
+
+  /// Event handler that fires every time an event is public in Pubsub.
+  js.CloudFunction onPublish(FutureOr<void> handler(PubsubEvent event)) {
+    dynamic wrapper(js.Event jsEvent) => _handleEvent(jsEvent, handler);
+    return nativeInstance.onPublish(allowInterop(wrapper));
+  }
+
+  /// Event handler that fires every time an event is public in Pubsub.
```

Comment:
This is not entirely accurate, feel free to remove this comment or update with more relevant information, up to you.